### PR TITLE
[datadog_synthetics_test] support json schema assertion in api and multi test

### PR DESCRIPTION
**Pull request by etnbrd**
This PR implements the support for JSON Schema validation of the body, for API and Multi-API synthetic tests.

### CODE REVIEW

File: datadog/resource_datadog_synthetics_test_.go [hunk 1]
```diff
@@ -479,6 +479,27 @@ func syntheticsAPIAssertion() *schema.Schema {
...
 				},
+				"targetjsonschema": {
+					Description: "Expected structure if `operator` is `validatesJSONSchema`. Exactly one nested block is allowed with the structure below.",
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"jsonschema": {
+								Description: "The Json Schema to validate the body against.",
```
Comment (Drarig29) [inline datadog/resource_datadog_synthetics_test_.go:490]: We use full uppercase `JSON` in other places, and it aligns with the [JSON Schema spec](https://json-schema.org/specification)
```suggestion
								Description: "The JSON Schema to validate the body against.",
```

File: datadog/resource_datadog_synthetics_test_.go [hunk 2]
```diff
@@ -479,6 +479,27 @@ func syntheticsAPIAssertion() *schema.Schema {
...
+								Type:        schema.TypeString,
+								Required:    true,
+							},
+							"metaschema": {
+								Description: "The meta Schema to use for the json Schema.",
```
Comment (Drarig29) [inline datadog/resource_datadog_synthetics_test_.go:495]:
```suggestion
								Description: "The meta-schema to use for the JSON Schema.",
```

File: docs/resources/synthetics_test.md
Comment (Drarig29) [inline docs/resources/synthetics_test.md:1]: (to be re-autogenerated)

**Review by Drarig29 — APPROVED** [review]
LGTM

**Review by HantingZhang2 — DISMISSED** [review]

**Review by alai97 — APPROVED** [review]
Looks good for docs!

**Review by HantingZhang2 — APPROVED** [review]

**Pull request merged**